vncviewer/MonitorArrangement.cxx

```diff
@@ -0,0 +1,220 @@
+/* Copyright 2021 Hugo Lundin <huglu@cendio.se> for Cendio AB.
+ * 
+ * This is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ * 
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * 
+ * You should have received a copy of the GNU General Public License
+ * along with this software; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307,
+ * USA.
+ */
+
+#include <vector>
+#include <algorithm>
+
+#include <FL/Fl.H>
+#include <FL/Fl_Window.H>
+#include <FL/Fl_Box.H>
+#include <FL/Fl_Button.H>
+#include <FL/Fl_Check_Button.H>
+#include <FL/Fl_Timer.H>
+#include <FL/Fl_Group.H>
+#include <FL/fl_draw.H>
+#include <FL/Fl_PNG_Image.H>
+#include <FL/Fl_Check_Browser.H>
+
+#include "MonitorArrangement.h"
+
+#define MONITOR_ARRANGEMENT_MARGIN 20
+#define MONITOR_MARGIN_SCALE_FACTOR 0.99
+#define MONITOR_AVAILABLE_COLOR fl_lighter(fl_lighter(fl_lighter(FL_BACKGROUND_COLOR)))
```

Comment:
This won't work as a `static const` though as it will be evaluated too early. I would prefer some other construction that keeps type checking though rather than using a `#define`.

Reply:
Moved into `MonitorArrangement` as const members.